codegen/smithy-ruby-codegen/src/main/java/software/amazon/smithy/ruby/codegen/config/ConfigProviderChain.java

```diff
@@ -109,7 +109,7 @@ public Builder staticProvider(String value) {
          * @return this builder
          */
         public Builder dynamicProvider(String rubyDefaultBlock) {
```

Comment:
Add documentation here to make it explicit that this will wrap the provided code in a proc that takes a single argument named "cfg" (and maybe even provide an example).  Otherwise I think this magic behavior would be easy to miss.